core/src/main/java/org/kiwix/kiwixmobile/core/utils/files/FileUtils.kt

```diff
@@ -228,4 +235,21 @@ object FileUtils {
   } catch (e: IOException) {
     "".also { e.printStackTrace() }
   }
+
+  @JvmStatic fun isValidFile(filePath: String): Boolean {
+    if (filePath.endsWith(".zim") || filePath.endsWith(".zimaa")) {
+      return true
+    }
+    return false
+  }
+
+  @JvmStatic fun getSdCardMainPath(context: Context): String {
+    var path = "${context.getExternalFilesDirs("")[1]}"
+    val separator: String = context.getString(R.string.android_directory_seperator)
+    val sepPos = path.indexOf(separator)
+    if (sepPos != -1) {
+      path = path.substring(0, sepPos)
+    }
```

Comment:
Kotlin String's `StringBefore` would clean this code.

Reply:
done